### PR TITLE
Add query building modes to time machines

### LINKED ISSUE #1500: add a button to switch to the script editor
as a query builder
with admin permissions or higher
when i view the explorer page
then i would like the ability to manually edit the query by hand
then i would like the page to remember my selection

### LINKED ISSUE #1501: add the panel for the visual query builder
as a query builder
with admin permissions or higher
when i view the query explorer
then i would like a place where i can select the elements of my query visually

### LINKED ISSUE #1502: add the ability to create multiple queries
as a query builder
with admin permissions or higher
when i view the explorer page
then i would like the ability to build out multiple queries visually

### LINKED ISSUE #1504: add the ability to create a new query
as a query builder
with admin permissions or higher
when i view the explorer tool
then i would like to be able to create a new query

by default, if not queries are there, we should open a new query explorer panel

### PR DESCRIPTION
**Pull request by chnn**
Closes #1500
Closes #1501
Closes #1502
Closes #1504

Stubs out a place for the visual query builder, multiple queries can be added/removed:

screen shot 2018-11-30 at 4 28 31 pm

A query can be edited either as Flux or InfluxQL:

screen shot 2018-11-30 at 4 28 35 pm

screen shot 2018-11-30 at 4 28 54 pm

screen shot 2018-11-30 at 4 29 17 pm

### CODE REVIEW

File: ui/src/shared/components/TimeMachineInfluxQLEditor.tsx [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+import {getActiveDraftScript} from 'src/shared/selectors/timeMachines'
+
+// Types
+import {AppState} from 'src/types/v2'
+
+const OPTIONS = {
+  mode: 'influxQL',
+  theme: 'influxql',
+  tabIndex: 1,
+  readonly: false,
```
Comment (mavarius) [inline ui/src/shared/components/TimeMachineInfluxQLEditor.tsx]: Should readOnly be camel case?

File: ui/src/shared/components/TimeMachineQueries.tsx [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+    queryStatus === RemoteDataState.Loading
+      ? ComponentStatus.Loading
+      : ComponentStatus.Default
+
+  return (
+    <div className="time-machine-queries">
+      <div className="time-machine-queries--controls">
+        <div className="time-machine-queries--tabs">
+          {range(queryCount).map(i => (
+            <TimeMachineQueryTab key={i} queryIndex={i} />
```
Comment (mavarius) [inline ui/src/shared/components/TimeMachineQueries.tsx]: Can we make the key more unique than an index number?

File: ui/src/shared/components/TimeMachineQueriesSwitcher.tsx [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+          InfluxQL
+        </Dropdown.Item>
+      </Dropdown>
+    )
+  }
+
+  private handleEditWithBuilder = (): void => {
+    const {onEditWithBuilder} = this.props
+
+    if (window.confirm(CONFIRM_LEAVE_ADVANCED_MODE)) {
```
Comment (mavarius) [inline ui/src/shared/components/TimeMachineQueriesSwitcher.tsx]: We'll need to replace this with a confirmation component in the application later.

File: ui/src/shared/components/TimeMachineInfluxQLEditor.tsx (hunk 1 above)
Comment (chnn) [reply ui/src/shared/components/TimeMachineInfluxQLEditor.tsx]: Woops, yep. Nice catch.

File: ui/src/shared/components/TimeMachineQueriesSwitcher.tsx (hunk 3 above)
Comment (chnn) [reply ui/src/shared/components/TimeMachineQueriesSwitcher.tsx]: Yeah what do you think would work. A `ConfirmButton` maybe?

File: ui/src/shared/components/TimeMachineQueries.tsx (hunk 2 above)
Comment (chnn) [reply ui/src/shared/components/TimeMachineQueries.tsx]: In this case the index is the sole identify for the tab (and is a stable identity), so I think it's ok.

**Pull request merged**